packages/chevrotain/src/parse/grammar/checks.ts

```diff
@@ -496,24 +489,23 @@ export function validateTooManyAlts(
   topLevelRule.accept(orCollector)
   const ors = orCollector.alternations
 
-  const errors = reduce(
-    ors,
-    (errors, currOr) => {
-      if (currOr.definition.length > 255) {
-        errors.push({
+  const errors = flatMap(ors, (currOr) => {
```

Comment:
as mentioned before, I think we are abusing flatMap when we combine it with an explicit filter.
the `reduce` verb makes more sense as it is a general transformation.
The most readable version imho would be two steps (filter + map)

Reply:
flatMap is the specific case of reduce on an array with the combination operation being concat. As this is exactly the case here, it is more readable to me.